apps/etterlatte-saksbehandling-ui/client/src/components/behandling/soeknadsoversikt/familieforhold/RedigerFamilieforhold.tsx

```diff
@@ -0,0 +1,155 @@
+import styled from 'styled-components'
+import { Alert, Button, Panel, TextField } from '@navikt/ds-react'
+import { InputList, InputRow } from '~components/person/journalfoeringsoppgave/nybehandling/OpprettNyBehandling'
+import React, { useState } from 'react'
+import { PlusIcon, XMarkIcon } from '@navikt/aksel-icons'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import { redigerFamilieforhold } from '~shared/api/behandling'
+import { isFailure, isPending, isSuccess } from '~shared/api/apiUtils'
+import { Personopplysninger, RedigertFamilieforhold } from '~shared/types/grunnlag'
+import { IDetaljertBehandling } from '~shared/types/IDetaljertBehandling'
+
+type Props = {
+  behandling: IDetaljertBehandling
+  personopplysninger: Personopplysninger
+}
+
+export const RedigerFamilieforhold = ({ behandling, personopplysninger }: Props) => {
+  const [redigerbartFamilieforhold, setFamilieforhold] = useState<RedigertFamilieforhold>({
+    gjenlevende: personopplysninger.gjenlevende.map((gjenlevende) => gjenlevende.opplysning.foedselsnummer),
+    avdoede: personopplysninger.avdoede.map((avdoede) => avdoede.opplysning.foedselsnummer),
+  })
+
+  const [status, redigerFamilieforholdRequest] = useApiCall(redigerFamilieforhold)
+  const [feilmelding, setFeilmelding] = useState<string | null>(null)
+
+  const lagre = () => {
+    setFeilmelding(null)
+    const foreldre = redigerbartFamilieforhold.avdoede.concat(redigerbartFamilieforhold.gjenlevende)
+    if (foreldre.length != 2) {
+      setFeilmelding('Mangler en eller flere forelder')
+    }
+    foreldre.find((fnr) => {
+      if (fnr.length !== 11) {
+        setFeilmelding('Ugyldig fødselsnummer')
+      }
+    })
+
+    if (feilmelding == null) {
+      redigerFamilieforholdRequest({
+        behandlingId: behandling.id,
+        redigert: redigerbartFamilieforhold,
+      })
+    }
```

Comment:
Det er dårlig praksis å sjekke state i samme funksjon som du setter den siden det i mange tilfeller kan medføre at du får et annet resultat enn du egentlig skulle. F.eks. kan du her få en feilmelding, uten at det er registrert når if-en sjekker på linje 38

Reply:
Selv om jeg reseter den i starten av funksjonen?
Hvis feilmelding skal vises etter "Lagre" så må de vel skje samtidig? Regner med at det ikke er så enkelt som å dele funksjonen i to?
Eneste andre alternativet er isåfall å validere løpende ettersom inputfeltene endres da? Men da får man jo feilmelding før man er ferdig med å fylle inn.

Reply:
Ja, det har ingenting å si siden useState hooken er async. Så denne funksjonen vil feile så og si alle ganger siden `feilmelding = null` og den ikke rekker å bli satt før `if (feilmelding...` slår til. 

Det beste er å gjøre noe i denne gaten: 

```
const feilmelding = noeSomFinnerFeil...

if (feilmelding)
   setFeilmelding
else 
   gjørDetDuSkal()
```

Da er du helt sikker på at feilmelding har forventet verdi før den sjekkes.